arbnode/node.go

```diff
@@ -348,7 +348,7 @@ func StakerDataposter(
 func createNodeImpl(
 	ctx context.Context,
 	stack *node.Node,
-	exec execution.FullExecutionClient,
```

Comment:
this really breaks execution split.
Instead - add the function you need to both FullExecutionNode on execution/interface, and to gethexec/ExecutionNode which implements this interface (it already has pointers to execution engine)

Reply:
Sure, I'll change it back, I agree with you.

I had read this https://go.dev/wiki/CodeReviewComments#interfaces and got mixed up in how I was thinking about it, but actually your current design matches what they recommend (interfaces not defined in same package providing the implementation).